### ISSUE TITLE
Fix parsing of unusual trailing expressions separated by whitespace

### ISSUE THREAD

**Issue by MaxGraey** [AssemblyScript/assemblyscript#1125]
The issue described in next examples:
```ts
function add(y: i32): i32 {
  let x = y 234;
  return x + y;
}

function foo(): i32 {
  return 123 456;
}
```
interpreted currently as:
```ts
function add(y: i32): i32 {
  let x = y;
  234; // skip as unreachable
  return x + y;
}

function foo(): i32 {
  return 123;
  456; // skip as unreachable
}
```
And normally compiled, but should be compiler error.

**Comment by MaxGraey** [comment]
Btw it should expect `,` after walk-through spaces and come across to new token which non-comma

**Comment by dcodeIO** [comment]
Looking at the issue again, a related aspect is that after a `return` it can be useful to not produce an error to aid conditional compilation. Imagine for example:

```ts
if (ASC_SOMEFLAG) {
  return;
}
doSomething();
```

where depending on the value of `ASC_SOMEFLAG` all subsequent code is unreachable, but it's intended. Not directly this issue, but could affect the given example of

```ts
function foo(): i32 {
  return 123;
  456; // skip as unreachable
}
```

in context of "but should be compiler error".

**Comment by MaxGraey** [comment]
Fixed in #2252
